src/modules/model.js

```diff
@@ -1,25 +1,29 @@
 const AWS = require('aws-sdk-wrap');
 const objectFields = require('object-fields');
-const { DefaultEntryNotFoundError, DefaultEntryExistsError } = require('./errors');
+const { DataMapper, DynamoDbSchema, DynamoDbTable } = require('@aws/dynamodb-data-mapper');
+const { DefaultItemNotFoundError, DefaultItemExistsError } = require('./errors');
 
-const DefaultEntryNotFound = ({ id }) => new DefaultEntryNotFoundError(id);
-const DefaultEntryExists = ({ id }) => new DefaultEntryExistsError(id);
-
-const generateExpressionAttributeNames = keys => Object.assign({}, keys
-  .reduce((p, k) => Object.assign(p, { [`#${k.toUpperCase()}`]: k }), {}), {});
+const DefaultItemNotFound = ({ id }) => new DefaultItemNotFoundError(id);
+const DefaultItemExists = ({ id }) => new DefaultItemExistsError(id);
 
 module.exports = ({
   modelName,
   tableName,
+  schema,
   awsConfig = {},
   errorMap: {
-    EntryNotFound = DefaultEntryNotFound,
-    EntryExists = DefaultEntryExists
+    ItemNotFound = DefaultItemNotFound,
+    ItemExists = DefaultItemExists
   } = {},
   callback = () => {}
 }) => {
+  class Model {}
```

Comment:
honestly looks like `id` should be part of the model constructor

Reply:
They internally instantiate the model when unmarshalling the data, I believe that's why they have a basic class in their docs. I personally believe we should stick to that pattern, but we can also wrap this in another class?
```
class Model {
  constructor(id) {
    class DataMapperClass {}
    this.DataMapperClass = DataMapperClass;
    ...
```